clang/lib/Analysis/CallGraph.cpp

```diff
@@ -152,20 +163,15 @@ class CGBuilder : public StmtVisitor<CGBuilder> {
 
   void VisitDeclStmt(DeclStmt *DS) {
     if (G->shouldSkipConstantExpressions()) {
-      auto IsConstexprVarDecl = [](Decl *D) {
-        if (const auto *VD = dyn_cast<VarDecl>(D))
-          return VD->isConstexpr();
-        return false;
-      };
-      if (llvm::any_of(DS->decls(), IsConstexprVarDecl)) {
-        assert(llvm::all_of(DS->decls(), IsConstexprVarDecl) &&
-               "Situation where a decl-group would be a mix of decl types, or "
-               "constexpr and not?");
-        return;
+      Decl **DeclPointer = DS->decls().begin();
+      for (Stmt *SubStmt : DS->children()) {
+        const Decl *D = *DeclPointer;
+        if (!isConstExprVar(*DeclPointer) && SubStmt)
```

Comment:
When *DeclPointer is nullptr, the routine returns false, do we want to visit the SubStmt?

Reply:
I don't think this case can exist.  If the `SubStmt ` exists, I think there will be a corresponding `Decl`. I'll change the `if (!D)` to an `assert` instead, so that its easier to catch this if I am mistaken.